src/icons/svg-icon-templates.ts

```diff
@@ -5,139 +5,568 @@
+                xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+                <title>search</title>
+                <path class="outer-shape" d="M16.33,5.05A10.95,10.95,0,1,1,5.39,16,11,11,0,0,1,16.33,5.05m0-2.05a13,13,0,1,0,13,13,13,13,0,0,0-13-13Z"/>
+                <path class="outer-shape" d="M35,33.29l-7.37-7.42-1.42,1.41,7.37,7.42A1,1,0,1,0,35,33.29Z"/>
+            </svg>
+        `,
+
+    "cog": `
+            <svg version="1.1" version="1.1" viewBox="0 0 36 36">
+                <title>cog</title>
+                <path class="inner-shape" d="M18.1,25c-3.9,0-7-3.1-7-7s3.1-7,7-7s7,3.1,7,7S22,25,18.1,25z M18.1,13c-2.8,0-5,2.2-5,5s2.2,5,5,5s5-2.2,5-5
+                    S20.9,13,18.1,13z"/>
+                <path class="outer-shape" d="M19.7,34h-3.4c-0.7,0-1.3-0.5-1.5-1.1L13.9,30c-0.5-0.2-1-0.4-1.5-0.6l-2.6,1.4c-0.6,0.3-1.4,0.2-1.9-0.3l-2.4-2.4
+                    c-0.5-0.5-0.6-1.3-0.3-1.9l1.4-2.6c-0.2-0.5-0.4-1-0.6-1.5l-2.8-0.9C2.5,21,2,20.4,2,19.7v-3.4c0-0.7,0.5-1.3,1.1-1.5l2.8-0.9
+                    c0.2-0.5,0.4-1.1,0.6-1.6L5.2,9.8C4.9,9.2,5,8.4,5.5,7.9l2.4-2.4C8.4,5,9.2,4.9,9.8,5.2l2.6,1.4c0.5-0.2,1-0.5,1.6-0.6l0.9-2.8
+                    C15,2.5,15.6,2,16.3,2h3.4c0.7,0,1.3,0.5,1.5,1.1l0.9,2.8c0.5,0.2,1,0.4,1.5,0.6l2.6-1.4c0.6-0.3,1.4-0.2,1.9,0.3l2.4,2.4
+                    c0.5,0.5,0.6,1.3,0.3,1.9l-1.4,2.6c0.2,0.5,0.4,1,0.6,1.5l2.8,0.9c0.7,0.2,1.2,0.8,1.2,1.6v3.4c0,0.7-0.5,1.3-1.1,1.5L30,22.1
+                    c-0.2,0.5-0.4,1-0.6,1.5l1.4,2.6c0.3,0.6,0.2,1.4-0.3,1.9l-2.4,2.4c-0.5,0.5-1.3,0.6-1.9,0.3l-2.6-1.4c-0.5,0.2-1,0.4-1.5,0.6
+                    l-0.9,2.9C21,33.5,20.4,34,19.7,34z M16.6,32h2.8l1.1-3.6l0.5-0.1c0.7-0.2,1.4-0.5,2.1-0.9l0.5-0.3l3.3,1.8l2-2l-1.8-3.3l0.3-0.5
+                    c0.4-0.7,0.6-1.4,0.9-2.1l0.1-0.5l3.6-1.1v-2.8l-3.6-1.1L28.3,15c-0.2-0.7-0.5-1.5-0.9-2.1l-0.3-0.5l1.8-3.3l-2-2l-3.3,1.8l-0.5-0.3
+                    c-0.7-0.4-1.4-0.7-2.1-0.9l-0.5-0.2L19.4,4h-2.8l-1.1,3.5L15,7.7c-0.8,0.2-1.5,0.5-2.2,0.9l-0.5,0.3L9.1,7.1l-2,2l1.7,3.3l-0.3,0.5
+                    c-0.4,0.7-0.7,1.4-0.9,2.2l-0.2,0.5L4,16.6v2.8l3.6,1.1L7.7,21c0.2,0.7,0.5,1.5,0.9,2.1l0.3,0.5l-1.8,3.3l2,2l3.3-1.8l0.5,0.3
+                    c0.7,0.4,1.4,0.6,2.1,0.9l0.5,0.1L16.6,32z"/>
+            </svg>
+        `,
+
+    get "settings" () {
+        return this.cog;
+    },
+
+    "share": `
+            <svg version="1.1" viewBox="0 0 36 36" preserveAspectRatio="xMidYMid meet"
+                xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+                <title>share</title>
+                <path class="outer-shape" d="M27.53,24a5,5,0,0,0-3.6,1.55L11.74,19.45a4.47,4.47,0,0,0,0-2.8l12.21-6.21a5.12,5.12,0,1,0-1.07-1.7L10.79,14.89a5,5,0,1,0,0,6.33l12.06,6.07A4.93,4.93,0,0,0,22.54,29a5,5,0,1,0,5-5Zm0-20a3,3,0,1,1-3,3A3,3,0,0,1,27.53,4ZM7,21a3,3,0,1,1,3-3A3,3,0,0,1,7,21ZM27.53,32a3,3,0,1,1,3-3A3,3,0,0,1,27.53,32Z"/>
```

Comment:
this is a compound path. we should avoid those and try to break it up into separate paths